dataactcore/utils/jobQueue.py

```diff
@@ -22,6 +30,32 @@ def enqueue(jobID):
     return response.json()
 
 
+def generate_f_file(submission_id, job_id, interface_holder_class,
+                    timestamped_name, is_local):
+    """Write rows from fileF.generateFRows to an appropriate CSV. Here the
+    third parameter, interface_holder_class, is a bit of a hack. Importing
+    InterfaceHolder directly causes cyclic dependency woes, so we're passing
+    in a class"""
+    with job_app.app_context():
+        job_manager = interface_holder_class().jobDb
+
+        try:
+            rows_of_dicts = fileF.generateFRows(job_manager.session,
+                                                submission_id)
+            header = [key for key in fileF.mappings]    # keep order
+            body = []
+            for row in rows_of_dicts:
+                body.append([row[key] for key in header])
+
+            write_csv(timestamped_name, is_local, header, body)
+            job_manager.markJobStatus(job_id, "finished")
```

Comment:
Putting these functions in the model classes seems like a good idea.  In the meantime, is there a benefit to moving them all into one file?  That seems like it would make it more difficult to find these functions, in the current organization they're at least split up by category of what they affect.  Maybe making them all static functions in the present locations could be a good first step, that way you wouldn't need an instance before calling the function.  Possibly some of the smaller functions could be eliminated and replaced with direct queries, however markJobStatus includes checking for jobs that have been marked as 'finished' and starting any dependent jobs that are ready.  We had talked about trying to have Celery manage dependencies, so if we get that working we could remove this function after that point.  Removing it prior to that would prevent validation and cross-file validation jobs from starting.